syntax = "proto3";

package thermalImagingDataCollect;

// The ThermalImagingDataCollect service definition.
service ThermalImagingDataCollectService {
    // Using a dataArray to render a ThermalImaging file
    rpc CollectThermalImagingData (ThermalImagingDataCollectRequest) returns (ThermalImagingDataCollectReply) {}
}

// The request message containing a tag for the server to log.
message ThermalImagingDataCollectRequest {
    string tag = 1;
}

// The response message containing the dataArray, and the errorMesage if any.
message ThermalImagingDataCollectReply {
    repeated double dataArray = 1;
    string errorMessage = 2;
}

// protoc -I grpc/ grpc/tidc.proto --go_out=plugins=grpc:grpc/
// python3 -m grpc_tools.protoc -I ./grpc/ --python_out=grpc --grpc_python_out=grpc ./grpc/tidc.proto